docs/DeveloperGuide.adoc

```diff
@@ -767,14 +809,316 @@ Use case ends.
++
+Use case ends.
+
+[discrete]
+=== Use case: Find person
+
+*MSS*
+
+1.  User requests to find one or more person(s) in AddressBook
+2.  AddressBook shows a list of persons with the name(s) specified by user
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. 0 person found
++
+Use case ends.
+
+[discrete]
+=== Use case: Add person
+
+*MSS*
+
+1.  User input command to add person
+2.  AddressBook adds the person
+3.  AddressBook displays message for successfully adding a person along with the person's details
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User input command wrongly
++
+[none]
+** 1a1. AddressBook shows an invalid format message and displays the right format to user
++
+Use case resumes at step 1.
+
+* 1b. User added a person already present in AddressBook
++
+[none]
+** 1b1. AddressBook shows duplicate person error message
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: List person(s) with specific tag
+
+*MSS*
+
+1.  User requests for a list of person(s) with a specified tag
+2.  AddressBook shows a list of person(s) with the tag specified by user
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. 0 person with such tag
++
+Use case ends.
+
+[discrete]
+=== Use case: Hide Selected Contacts
+
+*MSS*
+
+1.  User requests to a contact with the command: hide
+2.  With hide, AddressBook does not show hidden contacts
+3.  Hidden contacts will not show even when search command is invoked
+4.  User can check what he has hidden with: check hidden
+5.  User can undo hidden contacts with command: unhide
+6.  AddressBook displays message for successfully hiding contact(s) from view.
++
+Use Case ends.
+
+*Extensions*
+
+[none]
+* 1a. User can hide 1 contact with the command: hide name1
++
+Use case ends.
+
+[none]
+* 1b. User can hide multiple contacts with the command: hide >1
++
+[none]
+** 1b1. User can proceed to enter command: hide name1 name2
++
+Use case ends.
+
+[none]
+* 1c. User can hide contacts under a certain tag: hide exampleTagName
++
+Use case ends.
+
+[none]
+* 3a. AddressBook will show contacts that user has hidden in chronological order (reverse from present)
++
+Use case ends.
+
+[none]
+* 4a. User can undo hiding, similar to extension steps 1a, 1b, 1c, with command: unhide
++
+Use case ends.
+
+[discrete]
+=== Use case: Sort Contacts by Tag
+
+*Precondition: Some contacts in user's AddressBook memory has to already have tags under their contact details.*
+
+*MSS*
+
+1.  User requests to sort contacts by command: sort exampleOfTag1 exampleOfTag2
+2.  AddressBook displays contacts that have the respective tag(s) under contact details as initially edited by user
++
+Use case ends.
+
+[discrete]
+=== Use case: Delete Multiple Contacts at Once
+
+*MSS*
+
+1.  User requests to delete multiple contacts with command: del >1
+2.  AddressBook prompts user to enter multiple names to delete: del name1 name2 .. etc
+3.  Or user can choose to select names from AddressBook GUI to delete
```

Comment:
if it is an alternate path, do you think it is better to be listed under extension?